dnf/conf/substitutions.py

```diff
@@ -26,9 +26,10 @@
 
 class Substitutions(dict):
 
-    def __init__(self):
+    def __init__(self, arch=None):
```

Comment:
Just nothing? Personally, I'd merge just the change in `dnf\sack.py`. The rest can be thrown away.